pkg/controller/master/backup/backupTaget.go

```diff
@@ -91,7 +91,7 @@ func (h *TargetHandler) OnBackupTargetChange(key string, setting *harvesterv1.Se
 	target, err = h.validateTargetEndpoint(target)
 	if err != nil {
 		logrus.Errorf("invalid backup target, error: %s", err.Error())
-		return h.updateBackupTargetSetting(settingCpy, target, err)
+		return nil, err
```

Comment:
should return the error if it is an invalid config and leverage the [setting handler](https://github.com/harvester/harvester/blob/f5bde7fa462c09d0513b8f370f71f2dd974f298c/pkg/controller/master/setting/handler.go#L64-L69) to update the configured status properly.